var/spack/repos/builtin/packages/cppunit/package.py

```diff
@@ -13,3 +13,31 @@ class Cppunit(AutotoolsPackage):
     url = "http://dev-www.libreoffice.org/src/cppunit-1.13.2.tar.gz"
 
     version('1.13.2', '0eaf8bb1dcf4d16b12bec30d0732370390d35e6f')
+
+    variant('cxxstd',
+            default='default',
+            values=('default', '98', '11', '14', '17'),
+            multi=False,
+            description='Use the specified C++ standard when building.')
+
+    def setup_environment(self, spack_env, run_env):
+        cxxstd = self.spec.variants['cxxstd'].value
+        cxxstdflag = ''
```

Comment:
I think you can shorten this series of conditionals to:
```python
cxxstdflag = getattr(self.compiler, 'cxx{0}_flag'.format(cxxstd), '')
```

Reply:
You _could_, except that you now have a possible scenario where the author adds a new value (`20`, say) but has neglected to ensure that the top level `Compiler` class has a `cxx20_flag` attribute that will raise an exception if the subclass does not implement its own. In your case, this will be fail-silent. The following code:
``` python
        cxxstdflag = '' if cxxstdflag == 'default' else \
                     getattr(self.compiler, 'cxx{0}_flag'.format(cxxstd))
```
will deal with the `default` case as desired while allowing `getattr()` or the `Compiler` class system to raise an exception as appropriate in the case of an allowed value that cannot be dealt with sensibly. It's a little more verbose however, and might benefit from an explanatory comment besides. What do you think?

Reply:
> What do you think?

Fine with me if we want to be that defensive :+1:

Reply:
Addressed with https://github.com/spack/spack/commit/23e08a3909ddcf9d42f664044a183c35cdab7412.